### PR TITLE
ARROW-1682: [Doc] Expand S3/MinIO fileystem dataset documentation

### PR DESCRIPTION
**Pull request by fsaintjacques**

### CODE REVIEW

**Comment by alippai** [comment]
Does "for testing and benchmarking" mean that it's not optimal to store arrow/parquet files on Minio for production workloads?

**Comment by fsaintjacques** [comment]
I can't comment on the production quality of MinIO since I've never used it in such scenario. I meant this for reference to other developers who wants to test the S3 bindings without having to use an actual S3 bucket. For example, I have a dataset stored locally. When I want to test the difference between local and cloud storage, I spin a MinIO instance pointing to the same local directory and benchmark accordingly. toxiproxy is used to introduce latency and bandwidth limits mimicking S3.

**Comment by alippai** [comment]
Thanks, now I understand. So the pairing with toxiproxy is for the testing :))
That's what you wrote, I just misunderstood

File: docs/source/python/dataset.rst [hunk 1]
```diff
@@ -325,6 +325,22 @@ The currently available classes are :class:`~pyarrow.fs.S3FileSystem` and
 details.
 
 
+Reading from Minio
+------------------
+
+In addition to cloud storage, pyarrow also supports reading from a MinIO object
+storage instance emulating S3 APIs. Paired with toxiproxy, this is useful for
```
Comment (pitrou) [inline docs/source/python/dataset.rst:332]: Can you add hyperlinks to MinIO and toxyproxy?

File: docs/source/python/dataset.rst [hunk 2]
```diff
@@ -325,6 +325,22 @@ The currently available classes are :class:`~pyarrow.fs.S3FileSystem` and
...
+testing or benchmarking.
+
+.. code-block:: python
+
+    from pyarrow import fs
+
+    minio = fs.S3FileSystem(scheme="http", endpoint="localhost:9000")
```
Comment (pitrou) [inline docs/source/python/dataset.rst:339]: Add a comment that this assumes MinIO is running unencrypted on local port 9000?

**Pull request closed without merging**